compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp

```diff
@@ -398,7 +398,15 @@ struct FusionOfTensorOpsPass
             // Do not fuse producer generic op if it has more than one user.
             if (auto producerGenericOp =
                     dyn_cast<linalg::GenericOp>(producer)) {
-              return producerGenericOp->hasOneUse();
+              return producerGenericOp->hasOneUse() &&
+                     llvm::all_of(producerGenericOp.getIteratorTypesArray(),
+                                  linalg::isParallelIterator);
+            }
+
+            if (auto consumerGenericOp =
+                    dyn_cast<linalg::GenericOp>(producer)) {
+              return llvm::all_of(consumerGenericOp.getIteratorTypesArray(),
+                                  linalg::isParallelIterator);
             }
```

Comment:
Is it broken by https://github.com/llvm/llvm-project/commit/3f18f6a2cfecb080f006477c46d3626102841a17 ?

If so, @qedawkins could you help review?

Reply:
Yeah, this is intended actually. The changes upstream just expand what the pattern is capable of, but I don't think we want to do the kinds of fusions there by default just yet.

@Max191 this also needs to check that either the producer or consumer is a generic (we don't want to enable fusion-by-expansion of named ops just yet I don't believe).

Reply:
I talked to Quinn about this, and we figured we should start without changing anything, and we can try turning it on later and see if anything breaks.

Reply:
yeah what Max said. The plan is to actually use these patterns in another place first which will use a different control function.

Reply:
SGTM, can we update the comment as well?

Reply:
cc @MaheshRavishankar who is the codeowner